### PR TITLE
feat(useFocusTrap): support multiple refs

### PR DESCRIPTION
**Pull request by Sma11X**
### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

<details>
<summary><strong>⚠️ Slowing down new functions</strong></summary>
<br>

> **Warning**: **Slowing down new functions**
>
> As the VueUse audience continues to grow, we have been inundated with an overwhelming number of feature requests and pull requests. As a result, maintaining the project has become increasingly challenging and has stretched our capacity to its limits. As such, in the near future, we may need to slow down our acceptance of new features and prioritize the stability and quality of existing functions. **Please note that new features for VueUse may not be accepted at this time.** If you have any new ideas, we suggest that you first incorporate them into your own codebase, iterate on them to suit your needs, and assess their generalizability. If you strongly believe that your ideas are beneficial to the community, you may submit a pull request along with your use cases, and we would be happy to review and discuss them. Thank you for your understanding.

</details>

---

### Description

Support multiple refs and string array for useFocusTrap

### CODE REVIEW

File: packages/integrations/useFocusTrap/index.ts [hunk 1]
```diff
@@ -86,13 +87,16 @@ export function useFocusTrap(
     }
   }
 
+  const targets = computed(() => Array.isArray(target)
```
Comment (Alfred-Skyblue) [inline packages/integrations/useFocusTrap/index.ts:90]: We need to consider that `target` might be a `Ref<HTMLElement[]>`.

Comment (pkc918) [reply packages/integrations/useFocusTrap/index.ts:90]: cc: [multiple target](https://github.com/vueuse/vueuse/blob/main/packages/core/useIntersectionObserver/index.ts)

I think it should be the same as `useIntersectionObserver`

Comment (Sma11X) [reply packages/integrations/useFocusTrap/index.ts]: Updated, thanks for the tips!

File: packages/integrations/useFocusTrap/index.ts [hunk 2]
```diff
@@ -60,7 +61,7 @@ export interface UseFocusTrapReturn {
  * @see https://vueuse.org/useFocusTrap
  */
 export function useFocusTrap(
-  target: MaybeElementRef,
+  target: string | MaybeComputedElementRef | (MaybeComputedElementRef | string)[] | MaybeRefOrGetter<(MaybeComputedElementRef | string)[]>,
```
Comment (Alfred-Skyblue) [inline packages/integrations/useFocusTrap/index.ts:64]: Modifying the type to `Arrayable<MaybeRefOrGetter<string> | MaybeComputedElementRef>` seems more convenient.

Comment (Sma11X) [reply packages/integrations/useFocusTrap/index.ts]: I didn’t realize there was such a type, thanks for pointing it out.

**Review by antfu — APPROVED** [review]

**Pull request merged**